indico_toolkit/association/positioning.py

```diff
@@ -57,15 +57,15 @@ def positioned_above_overlap(
         """
         is_above = False
         is_min_overlap = True
-        if below_pos["page_num"] < above_pos["page_num"]:
+        if below_pos["page_num"] != above_pos["page_num"]:
             raise ToolkitInputError(
                 "Predictions are not on the same page!"
             )
-        if self.xaxis_overlap(above_pos, below_pos):
+        if self.xaxis_overlap(above_pos, below_pos) and self.yaxis_above(above_pos, below_pos):
             is_above = True
             horizontal_overlap_distance = abs(max(above_pos["bbLeft"], below_pos["bbLeft"]) - min(above_pos["bbRight"], below_pos["bbRight"]))
             position_width = abs(below_pos["bbLeft"] - below_pos["bbRight"])
-            if horizontal_overlap_distance / position_width < min_overlap_percent:
+            if horizontal_overlap_distance / position_width < min_overlap_percent and min_overlap_percent:
```

Comment:
you need to check min_overlap_percent first... the first condition will run and if min_overlap_percent is None it will raise a TypeError

Reply:
always raises issue that you must not have a test that accounts for this condition- should have a test to ensure no error is thrown where min_overlap_percent is None